core/GlobalState.cc

```diff
@@ -84,10 +87,9 @@ void GlobalState::initEmpty() {
     ENFORCE(id == Symbols::top());
     id = synthesizeClass(core::Names::Constants::Bottom(), 0);
     ENFORCE(id == Symbols::bottom());
-    id = synthesizeClass(core::Names::Constants::Root(), 0);
+    id = synthesizeClass(core::Names::Constants::Root(), 0, false, true);
```

Comment:
I really don't like reading function call sites like this, because I have no idea what `false` or `true` means.

Would you mind at least putting `false` and `true` in local variables, like

```cpp
auto isModule = false;
auto createSingleton = true;
id = synthesizeClass(core::Names::Constants::Root(), 0, isModule, createSingleton);
```

or even better create an `enum class` with two variants?

Up to you.